extension/intents/nicknames/nicknames.js

```diff
@@ -114,3 +102,89 @@ intentRunner.registerIntent({
     intentRunner.unregisterPageName(name);
   },
 });
+
+intentRunner.registerIntent({
+  name: "nicknames.pause",
+  async run(context) {
+    if (context.routineExecutor === undefined) {
+      const exc = new Error("Command not available.");
+      exc.displayMessage = "Command not available.";
+      throw exc;
+    }
+    browser.runtime.sendMessage({
+      type: "presentMessage",
+      message: context.slots.message,
+    });
+    context.routineExecutor.pauseRoutine();
+  },
+});
+
+intentRunner.registerIntent({
+  name: "nicknames.continue",
+  async run() {
+    const { pausedRoutine } = await browser.storage.local.get("pausedRoutine");
+    if (pausedRoutine === undefined) {
+      const exc = new Error(
+        "'Continue' can only be used in a routine after 'pause routine'"
+      );
+      exc.displayMessage =
+        "'Continue' can only be used in a routine after 'pause routine'";
+      throw exc;
+    }
+
+    const { name, nextIndex, states, forIndex } = pausedRoutine;
+    const registeredNicknames = await intentRunner.getRegisteredNicknames();
+    const routineExecutor = new RoutineExecutor(
+      registeredNicknames[name].nickname,
+      registeredNicknames[name].contexts,
+      nextIndex,
+      states,
+      forIndex
+    );
```

Comment:
This should definitely all happen in the RoutineExecutor. Pause/continue shouldn't have to know about index, states, or forIndex, these should all stay internal. Then as things are added they won't have to know about those things either.

You could use a `.toJSON()` method to serialize the executor, and `RoutineExecutor.hydrate(json)` to reinstantiate. But I don't think it needs to be kept in storage at all, paused routines can just live in a global variable.